packages/backend/src/lib/data/hooks/createIndexHook.js

```diff
@@ -0,0 +1,38 @@
+module.exports = function createIndexHook({
+  createIndexJob,
+  rebuildInProgress,
+  resource,
+}) {
+  return function indexHook({ item, serviceInteractor }) {
+    return Promise.resolve().then(() => {
+      const id = item && item.id
```

Comment:
should this hook bail if there's no id? (if item.id is guaranteed by validation to be here then we could skip the `item &&`)

Reply:
fixed